sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -203,35 +312,73 @@ joint_matrix_mad(Group sg, joint_matrix<T1, M, K, LayoutA, Group> &mA,
 }
 
 template <typename Group, typename T, size_t NumRows, size_t NumCols,
-          matrix_layout Layout, typename T2>
-inline __SYCL_ALWAYS_INLINE void
-joint_matrix_fill(Group sg,
-                  joint_matrix<T, NumRows, NumCols, Layout, Group> &res,
-                  const T2 v) {
+          matrix_use_enum Use, matrix_layout_enum Layout, typename T2>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_fill_base(
+    Group sg, joint_matrix_base<T, NumRows, NumCols, Layout, Use, Group> &res,
+    const T2 v) {
   // We kept the unused "sg" in joint_matrix_fill to match the other DPC++
   // functions
   (void)sg;
 #ifdef __SYCL_DEVICE_ONLY__
-  res.spvm =
-      __spirv_CompositeConstruct<T, NumRows, NumCols,
-                                 spv_matrix_layout_traits<Layout>::value>(
-          static_cast<T>(v));
+  res.spvm = __spirv_CompositeConstruct<T, NumRows, NumCols,
+                                        spv_matrix_layout_traits<Layout>::value,
+                                        spv_matrix_use_traits<Use>::value>(
+      static_cast<T>(v));
 
 #else
   (void)res;
   (void)v;
 #endif // __SYCL_DEVICE_ONLY__
 }
 
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          access::address_space Space, typename... Args>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_load(Group sg, joint_matrix<T, NumRows, NumCols, Args...> &res,
+                  multi_ptr<T, Space> src, size_t stride,
+                  matrix_layout_enum MemL) {
```

Comment:
@dkhaldi , i think there is issue here. since we consider matrix_layout::row_major as class instead of enumerate and a type can't play a role of function parameter, we can't pass memL in joint_matrix_load's func parameter. 
I haven't found solution yet. what we can do is just change into 'matrix_layout_enum' which is enumerate, but it is against users' previous habit.